src/mon/Monitor.cc

```diff
@@ -5293,6 +5293,23 @@ bool Monitor::ms_handle_refused(Connection *con)
   return false;
 }
 
+bool Monitor::ms_handle_throttle(THROTTLE_TYPE ttype) {
```

Comment:
This `ms_handle_throttle` should be in MonClient.

Reply:
I didn't do it because there is no active `clog` object (of type LogChannelRef) inside MonClient. I could use the `log_client` pointer which gets initialized by set_log_client(), but then I think I need to call log_client.create_channel() to do the cluster logging as done [here](https://github.com/ceph/ceph/blob/master/src/mon/Monitor.cc#L210). Instead I just handled it inside the Monitor Dispatcher, where a clog object already exists and that's straightforward. Let me know your thoughts. The other review comments are fixed.

Reply:
Putting the handler in the Monitor means that only the `ceph-mon` is handling throttling events. We want all daemons to do so.

You can check if `log_client != nullptr` before trying to setup a channel for writing to the clog. Daemons using MonClient will setup the log client for you. If they don't, then don't write anything to the clog.